src/_common/AppFadeCollapse.vue

```diff
@@ -0,0 +1,228 @@
+<script lang="ts" setup>
+import { ref } from '@vue/reactivity';
+import { onMounted, toRefs, watch } from 'vue';
+import { sleep } from '../utils/utils';
+import { Screen } from './screen/screen-service';
+import { Scroll } from './scroll/scroll.service';
+
+const ExtraCollapsePadding = 200;
+
+/**
+ * If the collapsed version would only leave this much pixels to expand it won't
+ * collapse.
+ */
+const Threshold = 50;
+const props = defineProps({
+	collapseHeight: {
+		type: Number,
+		required: true,
+	},
+	isOpen: {
+		type: Boolean,
+	},
+	animate: {
+		type: Boolean,
+	},
+	size: {
+		type: String,
+		required: false,
+		default: undefined,
+		validator: val => !val || val === 'sm',
+	},
+	ignoreThreshold: {
+		type: Boolean,
+	},
+	asMask: {
+		type: Boolean,
+	},
+});
+
+const { collapseHeight, isOpen, animate, size, ignoreThreshold, asMask } = toRefs(props);
+
+const emit = defineEmits({
+	requireChange: (_isRequired: boolean) => true,
+	expand: (_e: Event) => true,
+});
+
+const isCollapsed = ref(false);
+const isPrimed = ref(false); // private
```

Comment:
If there's variables that we don't need to watch, then just make them normal let variables. It's faster then.